### ISSUE TITLE
CF check_cell_methods failing incorrectly with repeated messages

### ISSUE THREAD

**Issue by mhidas** [ioos/compliance-checker#155]
[This file](http://thredds.aodn.org.au/thredds/catalog/IMOS/eMII/checker_test/ACORN/catalog.html?dataset=eMII/checker_test/ACORN/IMOS_ACORN_V_20140406T003000Z_ROT_FV00_1-hour-avg.nc) has a variable with the *cell_methods* attribute defined as

        float UCUR(TIME, LATITUDE, LONGITUDE) ;
                UCUR:cell_methods = "TIME: mean" ;

As far as I can tell from [CF](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.6/build/cf-conventions.html#cell-methods), this is correct. The output from the CF checker for this file includes:
```
var                                    :2:    29/39 :  
    UCUR                               :2:     3/ 4 :  
        cell_methods_name              :2:     0/ 1 : The name field does not
                                                      appear in the allowable
                                                      types., The name field
                                                      does not appear in the
                                                      allowable types., The name
                                                      field does not match the
                                                      dimension., The name field
                                                      does not match the
                                                      dimension., The name field
                                                      does not match the
                                                      dimension., The name field
                                                      does not match the
                                                      dimension., The name field
                                                      does not match the
                                                      dimension., The name field
                                                      does not match the
                                                      dimension., The name field
                                                      does not match the
                                                      reserved words "interval",
                                                      "area", or "comment"., The
                                                      name field does not match
                                                      the reserved words
                                                      "interval", "area", or
                                                      "comment".
```

**Comment by daf** [comment]
@mhidas the `cell_methods` checking appears to be totally broken.  Might get a chance to work on it today.

**Comment by mhidas** [comment]
@daf Yes, that was my impression too, but I didn't have time to try fixing it. In case it helps, I got as far as adding a unittest to catch this bug (perhaps the first of a few in there):
```python
    def test_check_cell_methods(self):
        dataset = self.get_pair(static_files['mapping'])
        results = self.cf.check_cell_methods(dataset)
        for r in results:
            self.assertTrue(r.value)
            self.assertFalse(r.msgs)
```

Let me know if you don't get a chance to work on it, and I'll have a go later this week.

**Comment by petejan** [comment]
The current code seems to fail when there is another : in the bracket, some experiments with regex...

What is a valid cell_method string? can I use this to check that the checker works?

LON: area TIME: average (interval: 10 days) LAT: std TIME3: stddev
(comment: 10) TEMP: area where land TEMP: area where land over sea TEMP:
area

Can use the following to find the Dimensions from the cell_methods,
~~~
\(.*?\)|(?P<var>\w*?:)
~~~
This is a good tool to check that it works, https://regex101.com/r/jI7jU7/2

Then use 
~~~
((?P<var>\w+): (?P<method>\w+) ?(?P<where>where (?P<wtypevar>\w+) ?(?P<over>over (?P<otypevar>\w+))?| ?)(?P<brace>\(((\w+): (\d+)(?P<unit>\w+)|(\w+): (\w+))\))*)
~~~
Check that it works here https://regex101.com/r/zN3aL3/2

to check if the cell_method is valid, then check var, method, wtypevar, otypevar, and unit against the valid values.

**Comment by daf** [comment]
This is fixed in 5510452, thanks @petejan !
